src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,903 @@
+					var notHorizontally = userSize.Width <= finalSize.Width;
+					var notVertically = userSize.Height <= finalSize.Height;
+
+					shouldAlign = (!notHorizontally && !notVertically, !notHorizontally && !notVertically);
+					break;
+
+				default:
+					// WinUI does not adjust alignment if the shape was smaller than the finalSize
+					shouldAlign = (userSize.Width > finalSize.Width, userSize.Height > finalSize.Height);
+					break;
+			}
+
+
+			var alignmentWidth = Math.Max(size.Width, renderingArea.Width);
+			var horizontalOverflow = alignmentWidth - finalSize.Width;
+			if (horizontalOverflow > 0 && shouldAlign.horizontally) 
+			{
+				switch (horizontal)
+				{
+					case HorizontalAlignment.Center:
+						renderingArea.X -= horizontalOverflow / 2.0;
+						break;
+
+					case HorizontalAlignment.Right:
+						renderingArea.X -= horizontalOverflow;
+						break;
+				}
+			}
+			var alignmentHeight = Math.Max(size.Height, renderingArea.Height);
+			var verticalOverflow = alignmentHeight - finalSize.Height;
+			if (verticalOverflow > 0 && shouldAlign.vertically)
+			{
+				switch (vertical)
+				{
+					case VerticalAlignment.Center:
+						renderingArea.Y -= verticalOverflow / 2.0;
+						break;
+
+					case VerticalAlignment.Bottom:
+						renderingArea.Y -= verticalOverflow;
+						break;
+				}
+			}
+
+			// For the Rectangle and the Ellipse, half of the StrokeThickness has to be excluded on each side of the shape.
+			var twoHalfStrokeThickness = StrokeThickness;
+			var halfStrokeThickness = twoHalfStrokeThickness / 2.0;
+			renderingArea.X += halfStrokeThickness;
+			renderingArea.Y += halfStrokeThickness;
+			renderingArea.Width -= twoHalfStrokeThickness;
+			renderingArea.Height -= twoHalfStrokeThickness;
+
+			return (size, renderingArea);
+		}
+
+#if !IS_DESIRED_SMALLER_THAN_CONSTRAINTS_ALLOWED
+		// The desired size before it has been changed to apply the [Min]<Width|Height>
+		private Size _realDesiredSize;
+#endif
+
+		private protected Size MeasureAbsoluteShape(Size availableSize, CGPath path)
+		{
+			if (path == null)
+			{
+				return default;
+			}
+
+			var stretch = Stretch;
+			var userSize = GetUserSizes();
+			var (userMinSize, userMaxSize) = GetMinMax(userSize);
+			var strokeThickness = StrokeThickness;
+			var halfStrokeThickness = GetHalfStrokeThickness();
+			var pathBounds = path.PathBoundingBox; // The BoundingBox does also contains bezier anchors even if out of geometry
+			var pathSize = (Size)pathBounds.Size;
+
+			if (nfloat.IsInfinity(pathBounds.Right) || nfloat.IsInfinity(pathBounds.Bottom))
+			{
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Ignoring path with invalid bounds {pathBounds}");
+				}
+
+				return default;
+			}
+
+			// Compute the final size of the Shape and the render properties
+			Size size;
+			(double x, double y) renderScale;
+			switch (stretch)
+			{
+				default:
+				case Stretch.None:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty 'case' clause.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5363624)